src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java

```diff
@@ -65,91 +195,411 @@
+    category.setShortDescription("Initial creation of case");
+
+    Case parentCase = createNewCase(sampleUnitParent, newCaseGroup);
+    if (sampleUnitParent.getSampleUnitChildren() != null) {
+      parentCase.setState(CaseState.INACTIONABLE);
+
+      for (SampleUnit sampleUnitChild :
+          sampleUnitParent.getSampleUnitChildren().getSampleUnitchildren()) {
+        Case childCase = createNewCase(sampleUnitChild, newCaseGroup);
+        caseRepo.saveAndFlush(childCase);
+        createCaseCreatedEvent(childCase, category);
+        log.with("case_id", childCase.getId().toString())
+            .with("sample_unit_type", childCase.getSampleUnitType().toString())
+            .debug("New Case created");
+      }
+    }
+    caseRepo.saveAndFlush(parentCase);
+    createCaseCreatedEvent(parentCase, category);
+    log.with("case_id", parentCase.getId().toString())
+        .with("sample_unit_type", parentCase.getSampleUnitType().toString())
+        .debug("New Case created");
+  }
+
+  /**
+   * Upfront fail fast validation - if this event is going to require a new case to be created,
+   * let's check the request is valid before we do something we cannot rollback ie IAC disable, or
+   * Action creation.
    *
-   * @param caseData the CaseCreation data
+   * @param category the category details
+   * @param oldCase the case the event is being created against
+   * @throws CTPException if the CaseEventRequest is invalid
    */
-  void createInitialCase(SampleUnitParent caseData);
+  private void validateCaseEventRequest(Category category, Case oldCase) throws CTPException {
+    String oldCaseSampleUnitType = oldCase.getSampleUnitType().name();
+    String expectedOldCaseSampleUnitTypes = category.getOldCaseSampleUnitTypes();
+    if (!compareOldCaseSampleUnitType(oldCaseSampleUnitType, expectedOldCaseSampleUnitTypes)) {
+      String errorMsg =
+          String.format(
+              WRONG_OLD_SAMPLE_UNIT_TYPE_MSG,
+              oldCaseSampleUnitType,
+              expectedOldCaseSampleUnitTypes);
+      log.error(errorMsg);
+      throw new CTPException(CTPException.Fault.VALIDATION_FAILED, errorMsg);
+    }
+  }
 
   /**
-   * Adds a new row to the caseiacaudit table with the case id and iac of given case
+   * To compare the old case sample unit type with the expected sample unit types
    *
-   * @param updatedCase the CaseCreation data
+   * @param oldCaseSampleUnitType the old case sample unit type
+   * @param expectedOldCaseSampleUnitTypes a comma separated list of expected sample unit types
+   * @return true if the expected types contain the old case sample unit type
    */
-  void saveCaseIacAudit(Case updatedCase);
+  private boolean compareOldCaseSampleUnitType(
+      String oldCaseSampleUnitType, String expectedOldCaseSampleUnitTypes) {
+    boolean result = false;
+    if (expectedOldCaseSampleUnitTypes != null) {
+      List<String> expectedTypes = Arrays.asList(expectedOldCaseSampleUnitTypes.split("\\s*,\\s*"));
```

Comment:
what the heck? Why are we having to split the string like this? Why would we store anything as a comma delimited list?

Reply:
I have absolutely no idea